### DISCUSSION TITLE
Save matrix A and right hand side b in MATLAB .mat format

### DISCUSSION THREAD

**Discussion by 436ahsan** [firedrakeproject/firedrake#2870] [Firedrake support] [answered]
Hi, I am trying to import assembled matrix A and the right hand side b of the linear system from Steady-state advection equation with up-winding problem (https://www.firedrakeproject.org/demos/upwind_advection.py.html) into MATLAB. Is there any way to save the matrix A and rhs b in .mat format or a python readable Compressed Sparse Row matrix format from firedrake? Thank you

**Comment by dham** [comment, marked as answer]
Courtesy of @pbrubeck on Slack:
```python
indptr, indices, values = assemble(a).petscmat.getValuesCSR()
import scipy
A = scipy.sparse.csr_matrix((values, (indptr, indices))
mdict = {"fem_matrix": A}
scipy.io.savemat("firedrake_output", mdict)
```
And:
```python
values = assemble(L).dat.data_ro
```
